src/devices/Hmc5883l/Hmc5883l.cs

```diff
@@ -0,0 +1,117 @@
+
+        /// <summary>
+        /// HMC5883L Direction Vector
+        /// </summary>
+        public Vector3 DirectionVector => ReadDirectionVector();
+
+        /// <summary>
+        /// HMC5883L Direction Angle (Deg)
+        /// </summary>
+        public double DirectionAngle => VectorToDirectionAngle(ReadDirectionVector());
+
+        /// <summary>
+        /// Initialize a new HMC5883L device connected through I2C
+        /// </summary>
+        /// <param name="sensor">I2C Device, like UnixI2cDevice or Windows10I2cDevice</param>
+        /// <param name="gain">Gain Setting</param>
+        /// <param name="measuringMode">The mode of measuring</param>
+        /// <param name="outputRate">Typical Data Output Rate (Hz)</param>
+        public Hmc5883l(I2cDevice sensor, Gain gain = Gain.Gain1090, MeasuringMode measuringMode = MeasuringMode.Continuous, OutputRate outputRate = OutputRate.Rate15)
+        {
+            _sensor = sensor;
+            _gain = (byte)gain;
+            _measuringMode = (byte)measuringMode;
+            _outputRate = (byte)outputRate;
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Initialize the sensor
+        /// </summary>
+        private void Initialize()
+        {
+            // Details in Datasheet P12
+            byte configA = (byte)(0x70 + _outputRate << 2);
+            byte configB = (byte)(_gain << 5);
+
+            _sensor.Write(new[] { (byte)Register.HMC_CONFIG_REG_A_ADDR, configA });
+            _sensor.Write(new[] { (byte)Register.HMC_CONFIG_REG_B_ADDR, configB });
+            _sensor.Write(new[] { (byte)Register.HMC_MODE_REG_ADDR, _measuringMode });
+        }
+
+        /// <summary>
+        /// Read raw data from HMC5883L
+        /// </summary>
+        /// <returns>Raw data</returns>
+        private Vector3 ReadDirectionVector()
+        {
+            Span<byte> xRead = stackalloc byte[2];
+            Span<byte> yRead = stackalloc byte[2];
+            Span<byte> zRead = stackalloc byte[2];
+
+            _sensor.WriteByte((byte)Register.HMC_X_MSB_REG_ADDR);
+            _sensor.Read(xRead);
+            _sensor.WriteByte((byte)Register.HMC_Y_MSB_REG_ADDR);
+            _sensor.Read(yRead);
+            _sensor.WriteByte((byte)Register.HMC_Z_MSB_REG_ADDR);
+            _sensor.Read(zRead);
+
+            short x = BinaryPrimitives.ReadInt16BigEndian(xRead);
+            short y = BinaryPrimitives.ReadInt16BigEndian(yRead);
+            short z = BinaryPrimitives.ReadInt16BigEndian(zRead);
+
+            return new Vector3(x, y, z);
+        }
+
+        /// <summary>
+        /// Calculate direction angle (Deg)
+        /// </summary>
+        /// <param name="directionVector">Hmc5883l raw data</param>
+        /// <returns>Angle</returns>
+        public double VectorToDirectionAngle(Vector3 directionVector)
+        {
+            double angle = Math.Atan2(directionVector.Y, directionVector.X) * (180 / Math.PI) + 180;
```

Comment:
I personally find radians much more useful in pretty much every use except for displaying on the screen